PYIC-8724: Update Manual F2F Pending Reset Lambda to throw errors on failure

AmritSidhu opened the pull request:
## Proposed changes
### What changed

- updates the manual-f2f-pending-reset Lambda so that execution failures are surfaced as red (failed) in the AWS Lambda console. 
- Now, when a failure occurs (e.g., missing user ID, no F2F record found, or an internal exception), the handler throws a RuntimeException, ensuring the Lambda execution result correctly reflects a failure.

<img width="600" height="500" alt="Screenshot 2025-10-15 at 13 22 46" src="https://github.com/user-attachments/assets/df46d778-8f66-439a-afd1-ecf85e925c16" />
<img width="600" height="500" alt="Screenshot 2025-10-15 at 13 23 00" src="https://github.com/user-attachments/assets/d4341b92-8565-4bed-9754-5808080ec4a7" />

### Why did it change

- Previously, even when an operation failed, the Lambda reported a “Succeeded” status because errors were returned in the response map instead of being thrown.

### Issue tracking
<!-- Jira ticket & other docs, like RFCs -->

- [PYIC-8724](https://govukverify.atlassian.net/browse/PYIC-8724)

## Checklists

- [ ] READMEs and documentation up-to-date
- [ ] API/ unit/ contract tests have been written/ updated
- [ ] No risk of exposure: PII, credentials, etc through logs/ code
- [ ] Production changes appropriately staged out


[PYIC-8724]: https://govukverify.atlassian.net/browse/PYIC-8724?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8724: Throw errors for failures

The manual-f2f-pending-reset Lambda currently shows all results in the same colour (green) in the interface, regardless of outcome. To reduce operator errors and align with the L1 user account deletion script, update the UI so that negative/error results are displayed in red instead of green. To do this we need to throw errors.

blakeyp commented on lambdas/manual-f2f-pending-reset/src/main/java/uk/gov/di/ipv/core/manualf2fpendingreset/ManualF2fPendingResetHandler.java:
```diff
@@ -85,14 +72,46 @@ public Map<String, Object> handleRequest(String input, Context context) {
+        }
+    }
+
+    private CriResponseItem findPendingRecord(String userId) {
```
Should this be void? we're returning the item but not actually using it in the calling code

AmritSidhu committed:
PYIC-8724: change findPendingRecord to void since return value unused

undermad commented on lambdas/manual-f2f-pending-reset/src/main/java/uk/gov/di/ipv/core/manualf2fpendingreset/ManualF2fPendingResetHandler.java:
```diff
@@ -85,14 +71,44 @@ public Map<String, Object> handleRequest(String input, Context context) {
+        } catch (Exception e) {
+            throw new ManualF2fPendingResetException("Failed to look up F2F record", e);
+        }
+    }
```
Hi Amrit, since this method doesn't return any value, maybe we should rename it to "checkIfPendingRecordExist" or something similar. Word "find" indicate it may return a value.

Also, this Throw e is somehow hard to understood and `getCriResponseItem` doesn't throw any exception

```
        var response = criResponseService.getCriResponseItem(userId, Cri.F2F);
        if(Objects.isNull(response)) {
            throw new ManualF2fPendingResetException("Failed to look up F2F record");
        }
```

undermad reviewed (commented):
Can we also bring back Arrange/Act/Assert to follow the pattern that we use in other places?

undermad commented on lambdas/manual-f2f-pending-reset/src/main/java/uk/gov/di/ipv/core/manualf2fpendingreset/ManualF2fPendingResetHandler.java:
```diff
@@ -85,14 +71,44 @@ public Map<String, Object> handleRequest(String input, Context context) {
+
+    private void deletePendingRecord(String userId) {
+        try {
+            criResponseService.deleteCriResponseItem(userId, Cri.F2F);
```
Same here, `deleteCriResponseItem` doesn't throw any exception so try catch is not required.
If for some reason we face any Exception it should be handled in the main handler, where we catch Exception and re-throw.

AmritSidhu committed:
PYIC-8724: Tidy up

Add arrange/act/assert comments to tests
Simplified deletePendingRecord since deleteCriResponseItem doesn't throw checked exceptions.
Renamed method to better reflect its purpose (validation, not retrieval) and
simplified the logic by removing redundant catch/rethrow structure.
